akka-stream/src/main/scala/akka/stream/impl/BoundedSourceQueue.scala

```diff
@@ -101,6 +107,10 @@ import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, Out
           case Done(QueueOfferResult.Failure(ex)) => failStage(ex)
           case NeedsActivation                    => throw new IllegalStateException // needs to be cleared before
         }
+
+      def notifySubscribers(): Unit = {
+        awaitingSubscribers.forEach(promise => promise.success(akka.Done.getInstance()))
```

Comment:
Here we notify **all** subscribers. Usually there should not be many (otherwise someone uses this `whenReady` improperly). We can think about notifying only a part of subscribers, but I don't feel this logic is really required. What do you think?